### PR TITLE
[VL] Update velox backend limitation docs for Spark3.4

### PR DESCRIPTION
**Pull request by ulysses-you**
## What changes were proposed in this pull request?

Update velox backend limitation docs since we have improved the related behavior for Spark3.4.

- We do not need overwrite spark class any more
- We support both static and dynamic table write and won't raise exception if not support

## How was this patch tested?

N/A

### CODE REVIEW

**Comment by ulysses-you** [comment]
cc @JkSelf @PHILO-HE thank you

File: docs/velox-backend-limitations.md [hunk 1]
```diff
@@ -6,7 +6,7 @@ nav_order: 5
 This document describes the limitations of velox backend by listing some known cases where exception will be thrown, gluten behaves incompatibly with spark, or certain plan's execution
 must fall back to vanilla spark, etc.
 
-### Override of Spark classes
+### Override of Spark classes (For Spark3.2 and Spark3.3)
```
Comment (PHILO-HE) [inline docs/velox-backend-limitations.md:9]: If not officially supported spark3.2/3.3 version is used,  `NoSuchMethodError` can be thrown at runtime, like the below issue reported. Could you help clarify it here? Thanks!
https://github.com/oap-project/gluten/issues/4514

Comment (ulysses-you) [reply docs/velox-backend-limitations.md:9]: sure, linked this issue

**Review by PHILO-HE — APPROVED** [review]
Looks good! Thanks!

**Comment by JkSelf** [comment]
@ulysses-you do we need to update the supported configurations in spark 34 [here](https://github.com/oap-project/gluten/blob/main/docs/velox-backend-limitations.md#configuration) ? It seems only support the `compression.codec`.

**Comment by ulysses-you** [comment]
@JkSelf it seems you have an on going pr for that.. so I leave it

**Pull request merged**